scarb/src/core/workspace.rs

```diff
@@ -56,6 +58,36 @@ impl<'c> Workspace<'c> {
     pub fn members(&self) -> impl Iterator<Item = Package> {
         [self.package.clone()].into_iter()
     }
+
+    pub fn current_profile(&self) -> Result<Profile> {
+        let profile = self.config.profile();
+        if profile.is_custom() && !self.current_package()?.has_profile(&profile) {
+            anyhow::bail!(
+                "package `{}` has no profile `{}`",
+                self.current_package()?,
+                profile
+            );
+        }
+        Ok(profile)
+    }
+
+    pub fn profile_names(&self) -> Result<Vec<String>> {
+        let mut names = self
+            .current_package()?
+            .manifest
+            .profiles
+            .iter()
+            .map(|p| p.to_string())
+            .collect::<Vec<_>>();
+        names.push(Profile::DEV.to_string());
+        names.push(Profile::RELEASE.to_string());
+        names.sort();
+        Ok(names)
+    }
+
+    pub fn compiler_config(&self) -> Result<ManifestCompilerConfig> {
+        Ok(self.current_package()?.manifest.compiler_config.clone())
+    }
```

Comment:
dead code? this function doesn't also make sense, because different workspace packages may have different compiler configs